plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaPageSink.java

```diff
@@ -126,7 +126,13 @@ public CompletableFuture<?> appendPage(Page page)
             keyBytes = keyEncoder.toByteArray();
             messageBytes = messageEncoder.toByteArray();
 
-            expectedWrittenBytes += keyBytes.length + messageBytes.length;
+            /**
+             * KafkeProducerBatch returns -1 for bytes length if data is null
```

Comment:
```suggestion
             * KafkaProducerBatch returns -1 for bytes length if data is null
```

Reply:
BTW this seems like an existing bugfix - what prevents the connector from writing empty messages today?

Reply:
The encoders used in this pr will return null, could be that the encoders already in use will just return 0, I will verify.

Reply:
This is an existing bugfix. If an empty message is written then bytes length will return 0, if it is null then it will return -1. This prevents the error from the check where the bytes written matches expected written bytes in the `finish` method.